### PR TITLE
Don't advice to modify default SCCs

### PR DESCRIPTION
**Pull request by php-coder**
From our experience, modifying default SCCs could lead to the issues during upgrade. While we told users that the only some fields can be safely modified, documentation had places where we editing properties of the "restricted" SCC that shouldn't not be modified as they will be reset to the default values by the `oc adm policy reconcile-sccs` command that is being executed during upgrade. 

This PR corrects documentation and advise to use other approaches instead of editing "restricted" SCC.

Related to https://bugzilla.redhat.com/show_bug.cgi?id=1577830

PTAL @pweil- @liggitt @openshift/sig-security

### CODE REVIEW

File: admin_guide/manage_scc.adoc
Comment (pweil-) [inline admin_guide/manage_scc.adoc]: without granting everyone access to more privileged SCCs such as privileged, hostaccess, or hostmount-anyuid.

**Review by pweil- — COMMENTED** [review]
LGTM with a minor wording change.  Thanks for updating it @php-coder

File: admin_guide/manage_scc.adoc
Comment (php-coder) [inline admin_guide/manage_scc.adoc]: This is not identical to editing `restricted` for the following reasons:
- `anyuid` has priority and will win other SCCs most of the time
- `anyuid` doesn't drop some UID-related capabilities that `restricted` drops
- `anyuid` also allows any `fsGroups`

I agree that granting `anyuid` to everyone could also lead to unexpected results and even break the cluster components in some cases (see https://github.com/openshift/openshift-ansible/pull/8558). But here, I think that it less painful than silent reset to the defaults during upgrade.

Ideally, I'd rather remove this chapter at all, because granting access for `system:authenticated` to any SCC may be a breaking change and IMHO it's a bad practice. Cluster admins should always create custom SCCs and grant access to them only to users/groups which are really need this access.

I decided to keep this chapter because it's about "how to go fast and ignore security best practices", so it's all big hack for someone who doesn't care/doesn't have time to fully understand/just making it work.

File: admin_guide/manage_scc.adoc
Comment (php-coder) [inline admin_guide/manage_scc.adoc]: This is also not identical change, because `nonroot` allows any fsgroups. Also there was a case, when using `nonroot` SCC broke integrated registry. So, It's hard to advise such thing but it the least of evils.

File: admin_guide/manage_scc.adoc
Comment (php-coder) [reply admin_guide/manage_scc.adoc]: @pweil- Thank you, Paul! You comment was addressed.

File: admin_guide/manage_scc.adoc
Comment (pweil-) [reply admin_guide/manage_scc.adoc]: add a NOTE section about this?

**Comment by bfallonf** [comment]
Thanks @php-coder . My original PR you link to was a simple solution.

Should we be waiting for someone from the security team to comment before we merge? Your comments about this section seem pretty concerning. Also, my original addition (The important box in the overview) is still there. Should that be deleted?

**Comment by kalexand-rh** [comment]
@enj, @simo5, PTAL

**Comment by php-coder** [comment]
> Also, my original addition (The important box in the overview) is still there. Should that be deleted?

I'd say that it should be modified at least to become a less strict.

File: admin_guide/manage_scc.adoc
Comment (php-coder) [reply admin_guide/manage_scc.adoc]: @pweil- I've added a WARNING section instead.

PTAL if you find a minute for me :)

File: admin_guide/manage_scc.adoc
Comment (simo5) [inline admin_guide/manage_scc.adoc]: SCC is being -> an SCC is
(?)

File: admin_guide/manage_scc.adoc
Comment (simo5) [inline admin_guide/manage_scc.adoc]: taking -> taken

File: admin_guide/manage_scc.adoc
Comment (simo5) [inline admin_guide/manage_scc.adoc]: accessible by -> granted to ?

File: admin_guide/manage_scc.adoc
Comment (simo5) [inline admin_guide/manage_scc.adoc]: In such a case ... -> This way potential issues will be confined to the affected users/groups and will not impact critical cluster components.

**Review by simo5 — CHANGES_REQUESTED** [review]
Minor language nits

File: admin_guide/manage_scc.adoc
Comment (pweil-) [inline admin_guide/manage_scc.adoc]: remove 'a' and 'an' from the component list

File: admin_guide/manage_scc.adoc
Comment (pweil-) [inline admin_guide/manage_scc.adoc]: when such SCC -> when an SCC

**Review by pweil- — COMMENTED** [review]
LGTM with minor nits.  Thanks!

**Comment by php-coder** [comment]
@simo5 @pweil- Thank you for the review! I've fixed everything that you have pointed out.

File: admin_guide/manage_scc.adoc
Comment (kalexand-rh) [inline admin_guide/manage_scc.adoc]: @php-coder, if this is still accurate, I'd use this text instead:

Be very careful with any modifications that have a cluster-wide impact. When you
grant an SCC to all authenticated users, as in the previous example, it also
affects Kubernetes and {product-title} components, including web console and
the integrated docker registry. Changes made with this SCC can cause these
components to stop functioning.

Take the same considerations into account when you modify an SCC, such as
granting the *restricted* SCC to everyone. Always prefer creating a custom
SCC and target it to only specific users or groups. This way potential issues
are confined to the affected users or groups and do not impact critical cluster
components.

Comment (kalexand-rh) [reply admin_guide/manage_scc.adoc]: Also, please take a look at the note at ~37 and make sure you're still happy with it.

File: admin_guide/manage_scc.adoc
Comment (kalexand-rh) [inline admin_guide/manage_scc.adoc]: s/*hostmount-anyuid*./*hostmount-anyuid*, take the following actions:

File: admin_guide/manage_scc.adoc
Comment (kalexand-rh) [inline admin_guide/manage_scc.adoc]: s/Create a new SCC (with name "hostpath", for example)/Create a new SCC:

Then please add the command to create the SCC.

File: admin_guide/manage_scc.adoc
Comment (kalexand-rh) [inline admin_guide/manage_scc.adoc]: s/Set `*allowHostDirVolumePlugin*` to `true`/Set the `*allowHostDirVolumePlugin*` parameter to `true` for the new SCC:

File: admin_guide/manage_scc.adoc
Comment (kalexand-rh) [inline admin_guide/manage_scc.adoc]: s/everyone/all users

File: admin_guide/manage_scc.adoc
Comment (kalexand-rh) [inline admin_guide/manage_scc.adoc]: s/volumes, should be/volumes are

**Review by kalexand-rh — COMMENTED** [review]
@php-coder, I have a few text edits and requests.

File: admin_guide/manage_scc.adoc
Comment (php-coder) [reply admin_guide/manage_scc.adoc]: I'd prefer to add a link to the chapter about SCC creation rather than providing this command again. WDYT?

Comment (kalexand-rh) [reply admin_guide/manage_scc.adoc]: Unless you expect people to be very familiar with creating an SCC, it's better to present them in the flow so they don't have to jump out. A link will work, though.

File: admin_guide/manage_scc.adoc
Comment (php-coder) [reply admin_guide/manage_scc.adoc]: > take a look at the note at ~37

I couldn't find this note. Is it a line number?

> Take the same considerations into account when you modify an SCC, such as
granting the restricted SCC to everyone.

This is not what I had in mind initially:

>> The same considerations should be taken into account when an SCC is being modified (for example, the *restricted* SCC is granted to everyone).

I've tried to say that when you're changing `restricted` SCC you have to think twice because this SCC is already granted to all users by default.

@kalexand-rh How I can express this idea then?

Comment (kalexand-rh) [reply admin_guide/manage_scc.adoc]: @php-coder, you're right. The note's at line 33 now:

> Do not modify the default SCCs. Customizing the default SCCs can lead to issues
when upgrading. Instead, xref:creating-new-security-context-constraints[create
new SCCs].

 Ok. How's this?
> Be very careful with any modifications that have a cluster-wide impact. When you
grant an SCC to all authenticated users, as in the previous example, or modify an
SCC that applies to all users, such as the _restricted_ SCC, it also affects
Kubernetes and {product-title} components, including the web console and
integrated docker registry. Changes made with this SCC can cause these
components to stop functioning.

> Always prefer creating a custom SCC and target it to only specific users or
groups. This way potential issues are confined to the affected users or groups
and do not impact critical cluster components.

**Comment by php-coder** [comment]
> you're right. The note's at line 33 now:

@kalexand-rh It's out of the scope of this PR at least. All others comments were addressed. Thank you and PTAL!

File: admin_guide/manage_scc.adoc
Comment (kalexand-rh) [inline admin_guide/manage_scc.adoc]: I have two more requests. :)

 s/Create a new SCCs] (with name "hostpath", for example)/[Create a new SCC] named `hostpath`:

Then squash your commits. Thanks!

**Comment by kalexand-rh** [comment]
@barleyer, will you PTAL or suggest another reviewer from QE?

@openshift/team-documentation, PTAL

File: admin_guide/manage_scc.adoc
Comment (php-coder) [reply admin_guide/manage_scc.adoc]: > I have two more requests. :)

No worries; both just have been pushed. Thanks!

File: admin_guide/manage_scc.adoc
Comment (mburke5678) [inline admin_guide/manage_scc.adoc]: @kalexand-rh This feels like a comma splice (before your changes, too!). Not sure how to fix.
"You can configure your cluster to not pre-allocate UIDs, which (?) allows containers to run as any user, but prevents privileged containers.  To configure this, grant access to the *anyuid* SCC for all users."
Something like that to offset  _grant access_ (above it almost sounds like granting access to everyone is another result of preventing pre-allocation.

File: admin_guide/manage_scc.adoc
Comment (mburke5678) [inline admin_guide/manage_scc.adoc]: s/Always prefer/Instead or Alternatively

File: admin_guide/manage_scc.adoc
Comment (mburke5678) [inline admin_guide/manage_scc.adoc]: s/take the following actions/perform the following ?

**Comment by mburke5678** [comment]
@kalexand-rh Couple of comments. Otherwise LGTM.

File: admin_guide/manage_scc.adoc
Comment (kalexand-rh) [reply admin_guide/manage_scc.adoc]: I don't think it's a comma splice, but the initial clause is way overloaded. I think it's trying to say, "To modify your cluster so that *all the things*, grant access to the *anyuid* SCC for everyone."

Maybe it should say:
```
When you grant access to the *anyuid* SCC for everyone, your cluster:
* Does not pre-allocate UIDs
* Allows containers to run as any user
* Prevents privileged containers
```

File: admin_guide/manage_scc.adoc
Comment (kalexand-rh) [reply admin_guide/manage_scc.adoc]: I'd s/Always prefer creating/It is recommended to create

File: admin_guide/manage_scc.adoc
Comment (kalexand-rh) [reply admin_guide/manage_scc.adoc]: maybe "perform the following actions"
The style guide talks about "Ambiguous pronoun references," and I think that making the object explicit improves clarity and follows the spirit of that rule. (It also makes it easier to translate, if we add more languages later.)

File: admin_guide/manage_scc.adoc
Comment (mburke5678) [reply admin_guide/manage_scc.adoc]: @kalexand-rh That looks better!

File: admin_guide/manage_scc.adoc
Comment (mburke5678) [reply admin_guide/manage_scc.adoc]: That seems to be a common construct in our docs.

Comment (bfallonf) [reply admin_guide/manage_scc.adoc]: "It is" is nearing expletive construction, and no need to say what's recommended when you can just say it. Maybe: 

"Instead, create a custom SCC and target specific users or groups so that potential issues are confined to those affected and do not impact critical cluster components."

**Comment by barleyer** [comment]
LGTM,besides I remember we have a card `Allow SCC Access via RBAC`，thanks.

**Comment by kalexand-rh** [comment]
@php-coder, it looks like this change might be valid back to 3.3. Does that sound right to you?

File: admin_guide/manage_scc.adoc
Comment (kalexand-rh) [reply admin_guide/manage_scc.adoc]: @bfallonf, "it is recommended" is the preferred upstream/downstream text in [the glossary.](https://doc-stage.usersys.redhat.com/documentation/en-us/ccs_internal_documentation/1.0/html-single/glossary_of_terms_and_conventions_for_product_documentation/#recommend)

**Comment by kalexand-rh** [comment]
Merging. I'll add the final text edits in a new PR.

**Comment by kalexand-rh** [comment]
Slava confirmed via email that this should go back to 3.3. I'll merge to master and then CP back to 3.3 on a PR that includes the last text edits.

**Comment by php-coder** [comment]
@kalexand-rh Thank you!

**Comment by kalexand-rh** [comment]
3.5: https://github.com/openshift/openshift-docs/pull/10168
3.4: https://github.com/openshift/openshift-docs/pull/10169
3.3: https://github.com/openshift/openshift-docs/pull/10170

**Pull request merged**
